core/auth/auth.go

```diff
@@ -139,15 +139,36 @@ func TokenAuth(cfg *config.Configuration) (http.RoundTripper, error) {
 
 // KeyAuth configures the key flow and returns an http.RoundTripper
 // that can be used to make authenticated requests using an access token
+// The KeyFlow requires a service account key and a private key.
+// If the private key is not provided explicitly, KeyAuth will check if there is one included
+// in the service account key and use that one.
 func KeyAuth(cfg *config.Configuration) (http.RoundTripper, error) {
 	err := getServiceAccountKey(cfg)
 	if err != nil {
 		return nil, fmt.Errorf("configuring key authentication: service account key could not be found: %w", err)
 	}
 
+	// Unmarshal service account key to check if private key is present
```

Comment:
The `KeyAuth` comment states (implicitly, that's how I read it at least) an explicitly passed private key takes precedence over the one in the service account key. But here it's the other way around.

Not sure which one is best. Usually I'd say to use the explicit option first, but here if the private key is given in the service account key then I don't see why that wouldn't match the service account key. So I think doing it as you're doing is fine, just clarify the comment

Reply:
The explicit private key takes precedence over the one in the service account key, I'll improve the comment.